packages/host/app/components/mode-switcher.gts

```diff
@@ -0,0 +1,158 @@
+        border-radius: var(--boxel-border-radius);
+        background: var(--boxel-purple-700);
+        color: var(--boxel-light);
+        font: 500 var(--boxel-font-sm);
+
+        position: relative;
+        display: flex;
+        justify-content: flex-start;
+        align-items: center;
+        width: 190px;
+        gap: var(--boxel-sp-sm);
+
+        --icon-color: var(--boxel-cyan);
+      }
+      .last-icon {
+        margin-left: auto;
+        
+        display: flex;
+      }
+      .content {
+        border-radius: var(--boxel-border-radius);
+        width: 190px;
+        background: rgba(0, 0, 0, 0.45);
+        color: var(--boxel-light);
+        font: 500 var(--boxel-font-sm);
+
+        --icon-color: var(--boxel-light);
+      }
+      :global(.ember-basic-dropdown-content) {
+        background: none;
+      }
+      :global(.content .boxel-menu__item) {
+        background: none;
+      }
+      :global(.content .boxel-menu__item:hover) {
+        background: rgba(0, 0, 0, 0.3);
+      }
+      :global(.content .boxel-menu__item > .boxel-menu__item__content) {
+        padding: var(--boxel-sp-xs);
+      }
+      :global(.content .menu-item) {
+        gap: var(--boxel-sp-sm);
+      }
+    </style>
+  </template>
+
+  modes: Mode[] = [
+    {
+      icon: 'eye',
+      label: 'Interact',
+    },
+    {
+      icon: 'icon-code',
+      label: 'Code',
+    }
+  ]
+  @tracked selectedMode: Mode = this.modes[0];
+  @service declare operatorModeStateService: OperatorModeStateService;
+  @service declare cardService: CardService;
+  @service private declare router: RouterService;
+  
+  @action 
+  buildMenuItem(onChoose: (mode: Mode) => void, mode: Mode): MenuItem {
+    return menuItemFunc([mode.label, () => onChoose(mode)], {icon: mode.icon});
+  }
+
+  @action
+  select(mode: Mode) {
+    if (this.selectedMode.label === mode.label) return;
+    this.selectedMode = mode;
+
+    switch (this.selectedMode.label) {
+      case 'Code':
+        this.openCodeMode.perform();
+        break;
+      case 'Interact':
+        this.openOperatorMode.perform();
+        break;
+    }
+  }
+
+  openCodeMode = restartableTask(async () => {
+    let topMostStackItems = this.operatorModeStateService.topMostStackItems();
+    let topMostStackItem;
+    let counter = 0;
+    do {
+      counter++;
+      topMostStackItem = topMostStackItems[topMostStackItems.length - counter];
+    } while (topMostStackItem.type !== 'card' && counter < topMostStackItems.length);
+    if (!topMostStackItem || topMostStackItem.type !== 'card') return;
+
+    let realmURL = await this.cardService.getRealmURL(topMostStackItem.card);
+    let path = topMostStackItem.card.id.replace(realmURL!.toString(), '') + '.json';
+    let pathArray = path.split('/');
+    pathArray.pop();
+    let openDirs = pathArray.length > 0 ? pathArray.join('/') + '/' : undefined;
+    this.router.transitionTo('code', { queryParams: { openDirs, path }});
```

Comment:
Is this supposed to transition into `code` route? From what I gather in [CS-5831](https://linear.app/cardstack/issue/CS-5831/code-mode-styled-3-column-layout-routing) it should toggle the `submode` param

Reply:
Yes, you're right, I didn't see this ticket [CS-5831](https://linear.app/cardstack/issue/CS-5831/code-mode-styled-3-column-layout-routing) before. I think I will take out the functionality from this PR.

Reply:
Updated [Take out functionality](https://github.com/cardstack/boxel/pull/554/commits/88d847e254393a30cb188bfdaa0fdb2c38b99809)